lib/pages/dashboard_page.dart

```diff
@@ -206,6 +208,9 @@ class _DashboardPageState extends State<DashboardPage> with WindowListener {
     });
 
     Future(() => _checkForUpdates(notifyIfLatest: false, notifyIfError: false));
+
+    _robotAlerts = RobotAlerts();
+    _robotAlerts.listen(ntConnection, context);
```

Comment:
Passing in the stateful widget's context to an non-mutable field is generally unsafe, and can cause several issues when the state of the app changes. Instead what I think should be done is instead of passing in the context where the listener class creates and displays the notification, there's a passed in method that takes in a parameter containing the data of the notification, and then the logic for creating and displaying the notification is handled in this method instead of in the notification listener class. A good example of this is the shuffleboard listener